gateway-service/src/main/java/org/zowe/apiml/gateway/security/mapping/model/MapperResponse.java

```diff
@@ -43,14 +43,18 @@ public String toString() {
     }
 
     public boolean isOIDCResultValid() {
-        // Some codes may be 4 and the result is still valid. But deny unless we know it for sure
         // https://www.ibm.com/docs/en/zos/2.5.0?topic=user-return-reason-codes
-        if (rc == 0 || safRc == 0 || racfRc == 0 || racfRs == 0) {
+        if (rc == 0 && safRc == 0 && racfRc == 0 && racfRs == 0) {
             return true;
         }
 
         if (rc == 8 && safRc == 8 && racfRc == 8) {
             switch (racfRs) {
+                case 4:
+                    apimlLog.log(OIDC_FAILED_MESSAGE_KEY,
+                        "A parameter list error occurred. Make sure LU01316 PTF was applied when using" +
+                            " the ACF2 security manager. Otherwise, contact Zowe support.");
+                    return false;
```

Comment:
kind of obvious question from me but this combination of return codes is sure to happen only on ACF2?

Reply:
No other ESM has this know issue, right? I guess it is ok, at least till we do not have available information on which ESM is in use.

Reply:
This message means that we are incorrectly calling SAF API. There is nothing that the user can do to fix it. However, ACF2 also contains a bug that leads to this error.

So the user can either apply a fix or it is our problem.